Source/Fuse.Reactive.Expressions/BinaryOperator.uno

```diff
@@ -68,24 +74,39 @@ namespace Fuse.Reactive
 			{
 				_leftSub = _bo.Left.Subscribe(context, this);
 				_rightSub = _bo.Right.Subscribe(context, this);
+				UpdateOperands();
 			}
 
 			protected override void OnNewData(IExpression source, object value)
 			{
 				if (source == _bo.Left) { _hasLeft = true; _left = value; }
 				if (source == _bo.Right) { _hasRight = true; _right = value; }
-
-				if ((_hasLeft || _bo.IsLeftOptional) && (_hasRight || _bo.IsRightOptional))
-					OnNewOperands(_left, _right);
+				UpdateOperands();
 			}
-
-			protected virtual void OnNewOperands(object left, object right)
+			
+			protected override void OnLostData(IExpression source)
+			{
+				if (source == _bo.Left) { _hasLeft = false; _left = null; }
+				if (source == _bo.Right) { _hasRight = false; _right = null; }
+				UpdateOperands();
+			}
+			
+			void UpdateOperands()
 			{
 				ClearDiagnostic();
-
+				
```

Comment:
Inserting a bunch of tabs here seems a bit pointless...

Reply:
I didn't actually. This is just how git diff decided to render the changes I made. I actually moved a whole block of code to after the ClearDiagnostic in a new function. The same with the other places where it looks like I needlessly added tabs.  It's just git assuming two lines are the same which really were in different places.

Reply:
OK, fair enough!